lib/ts/recipe/multifactorauth/api/implementation.ts

```diff
@@ -25,89 +25,74 @@ export default function getAPIInterface(): APIInterface {
-                const allAvailableSecondaryFactors = await options.recipeInstance.getAllAvailableSecondaryFactorIds(
-                    mfaInfo.tenantConfig
-                );
+            const factorsAlreadySetup = mfaInfo.factorsSetUpForUser;
+            const allAvailableSecondaryFactors = await options.recipeInstance.getAllAvailableSecondaryFactorIds(
+                mfaInfo.tenantConfig
+            );
 
-                const factorsAllowedToSetup: string[] = [];
-                for (const id of allAvailableSecondaryFactors) {
-                    try {
-                        await options.recipeImplementation.assertAllowedToSetupFactorElseThrowInvalidClaimError({
-                            session,
-                            factorId: id,
-                            factorsSetUpForUser: factorsAlreadySetup,
-                            mfaRequirementsForAuth: mfaInfo.mfaRequirementsForAuth,
-                            userContext,
-                        });
-                        factorsAllowedToSetup.push(id);
-                    } catch (err) {
-                        // ignore
-                    }
-                }
-
-                const nextSetOfUnsatisfiedFactors = MultiFactorAuthClaim.getNextSetOfUnsatisfiedFactors(
-                    mfaInfo.completedFactors,
-                    mfaInfo.mfaRequirementsForAuth
-                );
-
-                let getEmailsForFactorsResult = options.recipeInstance.getEmailsForFactors(
-                    mfaInfo.sessionUser,
-                    session.getRecipeUserId()
-                );
-                let getPhoneNumbersForFactorsResult = options.recipeInstance.getPhoneNumbersForFactors(
-                    mfaInfo.sessionUser,
-                    session.getRecipeUserId()
-                );
-                if (
-                    getEmailsForFactorsResult.status === "UNKNOWN_SESSION_RECIPE_USER_ID" ||
-                    getPhoneNumbersForFactorsResult.status === "UNKNOWN_SESSION_RECIPE_USER_ID"
-                ) {
-                    throw new SessionError({
-                        type: "UNAUTHORISED",
-                        message: "User no longer associated with the session",
-                        payload: {
-                            clearTokens: true,
-                        },
+            const factorsAllowedToSetup: string[] = [];
+            for (const id of allAvailableSecondaryFactors) {
+                try {
+                    await options.recipeImplementation.assertAllowedToSetupFactorElseThrowInvalidClaimError({
+                        session,
+                        factorId: id,
+                        factorsSetUpForUser: factorsAlreadySetup,
+                        mfaRequirementsForAuth: mfaInfo.mfaRequirementsForAuth,
+                        userContext,
                     });
+                    factorsAllowedToSetup.push(id);
+                } catch (err) {
+                    // ignore
```

Comment:
check for the case if assertAllowedToSetupFactorElseThrowInvalidClaimError throws the claim error, and only then ignore. Else throw err

Reply:
Done